.taskcluster.yml

```diff
@@ -5,158 +5,194 @@ version: 1
 policy:
   pullRequests: public
 tasks:
-  - $if: 'tasks_for == "github-pull-request" && event["action"] in ["opened", "reopened", "synchronize"]'
-    then:
+  $let:
+    decision_task_id: {$eval: as_slugid("decision_task")}
+    expires_in: {$fromNow: '1 year'}
+    user: ${event.sender.login}
+
+    # We define the following variable at the very top, because they are used in the
+    # default definition
+    head_branch:
+      $if: 'tasks_for == "github-pull-request"'
+      then: ${event.pull_request.head.ref}
+      else: ${event.release.target_commitish}
+
+    head_rev:
+      $if: 'tasks_for == "github-pull-request"'
+      then: ${event.pull_request.head.sha}
+      else: ${event.release.tag_name}
+
+    repository:
+      $if: 'tasks_for == "github-pull-request"'
+      then: ${event.pull_request.head.repo.html_url}
+      else: ${event.repository.html_url}
+
+    scheduler_id:
+      $if: 'tasks_for == "cron"'
+      then: focus-nightly-sched  # TODO: Rename to mobile-nightly-sched
+      else: taskcluster-github
+
+    is_repo_trusted:
+      # Pull requests on main repository can't be trusted because anybody can open a PR on it, without a review
+      $if: 'tasks_for in ["github-push", "github-release", "cron"] && event.repository.html_url == "https://github.com/mozilla-mobile/reference-browser"'
+      then: true
+      else: false
+  in:
+    $let:
+      decision_worker_type:
+        $if: 'is_repo_trusted'
```

Comment:
Yeah :/ I don't think this is a big deal for us, yet. So far, we just need 2 layers. It would become a bigger deal if we had to deal with 4, I think.

Reply:
It's unnecessary noise that you need to filter out when scanning through the file, and it throws you off: when I see a block is indented, I implicitly assume that it's a bit "different" from less-indented blocks. Though we haven't reached a "staircase" (:stuck_out_tongue:) amount of steps for `$let` just yet, it's still a little jarring to have to filter out that "yes this is indented, but it's just more var declarations"